tests_metricflow/sql/optimizer/test_cte_table_alias_simplifier.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import annotations
+
+import pytest
+from _pytest.fixtures import FixtureRequest
+from metricflow_semantics.sql.sql_join_type import SqlJoinType
+from metricflow_semantics.sql.sql_table import SqlTable
+from metricflow_semantics.test_helpers.config_helpers import MetricFlowTestConfiguration
+
+from metricflow.sql.optimizer.table_alias_simplifier import SqlTableAliasSimplifier
+from metricflow.sql.render.sql_plan_renderer import DefaultSqlQueryPlanRenderer
+from metricflow.sql.sql_exprs import (
+    SqlColumnReference,
+    SqlColumnReferenceExpression,
+    SqlComparison,
+    SqlComparisonExpression,
+)
+from metricflow.sql.sql_plan import (
+    SqlCteNode,
+    SqlJoinDescription,
+    SqlSelectColumn,
+    SqlSelectStatementNode,
+    SqlTableNode,
+)
+from tests_metricflow.sql.optimizer.check_optimizer import assert_optimizer_result_snapshot_equal
+
+
+@pytest.fixture
+def sql_plan_renderer() -> DefaultSqlQueryPlanRenderer:  # noqa: D103
+    return DefaultSqlQueryPlanRenderer()
+
+
+def test_table_alias_simplification(
+    request: FixtureRequest,
+    mf_test_configuration: MetricFlowTestConfiguration,
+    sql_plan_renderer: DefaultSqlQueryPlanRenderer,
+) -> None:
+    """Tests that table aliases are removed when not needed in CTEs."""
```

Comment:
Should we also have a test to demonstrate that table aliases are not removed if they are needed in CTEs?
I'm assuming that would only happen if there was a join within the CTE statement. Not sure if that case will ever occur with the current CTE implementation.

Reply:
Can't hurt - added a test case.